### PR TITLE
Add capability for deleted_on, deleted_by attributes

### PR DESCRIPTION
**Pull request by nareshkharola123**
## Description

Added capability for deleted_on, deleted_by attributes. 
dependency that is required: loopback4-authentication

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested ?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [x] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the style guide
- [ ] API Documentation in code was updated

### CODE REVIEW

File: src/models/soft-delete-entity.ts [hunk 1]
```diff
@@ -15,6 +15,20 @@ export abstract class SoftDeleteEntity extends Entity {
...
+  @property({
+    type: 'date',
+    default: false,
+    name: 'deleted_on',
+  })
+  deletedOn?: Date;
+
+  @property({
+    type: 'number',
+    default: false,
```
Comment (samarpan-b) [inline src/models/soft-delete-entity.ts:27]: mark this nullable and no default

File: src/models/soft-delete-entity.ts [hunk 2]
```diff
@@ -15,6 +15,20 @@ export abstract class SoftDeleteEntity extends Entity {
   })
   deleted?: boolean;
 
+  @property({
+    type: 'date',
+    default: false,
```
Comment (samarpan-b) [inline src/models/soft-delete-entity.ts:20]: mark this nullable and no defaults

File: src/repositories/soft-crud.repository.base.ts [hunk 3]
```diff
@@ -23,6 +26,8 @@ export abstract class SoftCrudRepository<
       prototype: T;
     },
     dataSource: juggler.DataSource,
+    @inject(AuthenticationBindings.CURRENT_USER)
+    private readonly user: AuthUser | undefined,
```
Comment (samarpan-b) [inline src/repositories/soft-crud.repository.base.ts:30]: use IauthUser interface here

File: src/repositories/soft-crud.repository.base.ts [hunk 4]
```diff
@@ -274,4 +287,13 @@ export abstract class SoftCrudRepository<
     // Do hard delete
     return super.deleteById(id, options);
   }
+
+  private getUserId(): number | undefined {
```
Comment (samarpan-b) [inline src/repositories/soft-crud.repository.base.ts:291]: why this ?

File: src/models/soft-delete-entity.ts (hunk 2 above)
Comment (nareshkharola123) [reply src/models/soft-delete-entity.ts]: done

File: src/repositories/soft-crud.repository.base.ts (hunk 3 above)
Comment (nareshkharola123) [reply src/repositories/soft-crud.repository.base.ts]: done

File: src/models/soft-delete-entity.ts (hunk 1 above)
Comment (nareshkharola123) [reply src/models/soft-delete-entity.ts]: done

**Comment by danysz** [comment]
Anyone is going to take care about this and merge it ?
Thanks

**Comment by samarpanB** [comment]
It will be merged and published today

**Comment by danysz** [comment]
> It will be merged and published today

Amazing.. I am starting a project and I really need this.

**Comment by danysz** [comment]
> It will be merged and published today

still in plan ?

**Comment by danysz** [comment]
Thanks

**Pull request merged**